src/autoload/global.gd

```diff
@@ -113,69 +118,76 @@ func _ready() -> void:
-		push_warning("Config file was corrupted! Using default settings")
+	if not config_file.has_section(CONFIG_SECTION):
+		push_warning("Config missing section %s, using default settings" % CONFIG_SECTION)
 		return
-	var new_config := new_config_variant as Dictionary
-
-	config = new_config.duplicate()
 
-	for entry in DEFAULT_CONFIG:
-		if entry not in config:
-			config[entry] = DEFAULT_CONFIG[entry]
+	for key in config:
+		if config_file.has_section_key(CONFIG_SECTION, key):
+			var value_variant = config_file.get_value(CONFIG_SECTION, key)
+			if typeof(value_variant) == typeof(DEFAULT_CONFIG[key]):
+				if key in ["music_volume", "sound_volume"]:
+					value_variant = clampf(value_variant, 0.0, 1.0)
+				elif key == "anim_key_idx":
+					value_variant = clampi(value_variant, 0, ANIMATION_SPEEDS.size() - 1)
+				config[key] = value_variant
+
+	# Do we already have a custom window size? Via --resolution, having an "override", etc
+	var intended_window_size := Vector2i(
+			ProjectSettings.get_setting("display/window/size/viewport_width"),
+			ProjectSettings.get_setting("display/window/size/viewport_height")
+	)
+	var current_window_size := get_window().size
+	var already_custom_size := current_window_size != intended_window_size
 
 	# Set the size of the window and center it.
-	if "window_size" in config:
-		var raw_size := config["window_size"] as Vector2i
-		var fixed_size: Vector2i
-		if 4 * raw_size.x > 3 * raw_size.y:
-			# Wider than it is supposed to be, use the height as the guide.
-			fixed_size = Vector2i(snapped(raw_size.y, 3) * 4 / 3, snapped(raw_size.y, 3))
-		else:
-			# Taller than it is supposed to be, use the width as the guide.
-			fixed_size = Vector2i(snapped(raw_size.x, 4), snapped(raw_size.x, 4) * 3 / 4)
-		get_window().size = fixed_size
-		var screen_id := get_window().current_screen
-		var screen_center := DisplayServer.screen_get_position(screen_id) \
-				+ DisplayServer.screen_get_size(screen_id) / 2
-		get_window().position = screen_center - fixed_size / 2
-
-	if config.get("fullscreen", false):
+	if (not already_custom_size):
+		var raw_size_variant = config["window_size"]
+		if typeof(raw_size_variant) == TYPE_VECTOR2I and raw_size_variant != intended_window_size:
+			var raw_size: Vector2i = raw_size_variant
+			var fixed_size: Vector2i
+			if 4 * raw_size.x > 3 * raw_size.y:
+				# Wider than it is supposed to be, use the height as the guide.
+				fixed_size = Vector2i(snapped(raw_size.y, 3) * 4 / 3, snapped(raw_size.y, 3))
+			else:
+				# Taller than it is supposed to be, use the width as the guide.
+				fixed_size = Vector2i(snapped(raw_size.x, 4), snapped(raw_size.x, 4) * 3 / 4)
+			get_window().size = fixed_size
+			var screen_id := get_window().current_screen
+			var screen_center := DisplayServer.screen_get_position(screen_id) \
+					+ DisplayServer.screen_get_size(screen_id) / 2
+			get_window().position = screen_center - fixed_size / 2
+
+	if config["fullscreen"]:
```

Comment:
Should this be inside the if? I could be wrong but I think as is, if we specify a resolution using the command line, but this flag is set, it will override that.

Reply:
It shouldn't be in the if statement to prevent the config from diverging from the actual fullscreen state. But I added an `else` clause to set fullscreen to false if you specify a `--resolution`